Tests/SentryTests/Transaction/SentryTracerTests.swift

```diff
@@ -378,27 +376,27 @@ class SentryTracerTests: XCTestCase {
         XCTAssertEqual(1, debugImageProvider.getDebugImagesFromCacheForFramesInvocations.count, "Tracer must retrieve debug images from cache.")
     }
 
-    func testDeadlineTimer_OnlyForAutoTransactions() {
+    func testDeadlineTimer_OnlyForAutoTransactions() throws {
         let sut = fixture.getSut(idleTimeout: fixture.idleTimeout)
         let child1 = sut.startChild(operation: fixture.transactionOperation)
         let child2 = sut.startChild(operation: fixture.transactionOperation)
         let child3 = sut.startChild(operation: fixture.transactionOperation)
 
         child3.finish()
 
-        fixture.timerFactory.fire()
+        try fixture.timerFactory.fire()
 
-        XCTAssertEqual(sut.status, .undefined)
-        XCTAssertEqual(child1.status, .undefined)
-        XCTAssertEqual(child2.status, .undefined)
+        XCTAssertEqual(sut.status, .deadlineExceeded)
+        XCTAssertEqual(child1.status, .deadlineExceeded)
+        XCTAssertEqual(child2.status, .deadlineExceeded)
```

Comment:
So I am unsure of what is going on here, or what the correct result should be. Previously, the call to fire was a no-op, see my above comment on the dispatch queue wrapper configuration. Now that the deadline timer is actually configured, this is the new result. I traced execution from the call to child3.finish() down into SentryTracer.canBeFinished, and it winds up taking the branch here: https://github.com/getsentry/sentry-cocoa/blob/44332bfdab0ae43dbe4edd86cd2d72de4fab4507/Sources/Sentry/SentryTracer.m#L506-L512, so the remaining spans are still in flight when the deadline timer is fired on the next line of the test.

Reply:
Good catch, I pushed a commit to address this: https://github.com/getsentry/sentry-cocoa/pull/4826/commits/e36a8ce8a3c052772566884d9e8bc48b02c1214a.